src/abstractdataframe/selection.jl

```diff
@@ -1578,13 +1574,14 @@ function _manipulate(df::AbstractDataFrame, normalized_cs::Vector{Any}, copycols
                         end
                     end
                     # here even if keeprows is true all is OK
-                    newdf[!, newname] = copycols ? df[:, i] : df[!, i]
+                    newdf[!, newname] = copycols || source_col_used[i] ? df[:, i] : df[!, i]
+                    copycols || (source_col_used[i] = true)
```

Comment:
Would this be equivalent? Could be clearer (we just keep track of whether the column was used or not). If so, the same change could be made above.
```suggestion
                    source_col_used[i] = true
```

Reply:
I will do something different. I will remove `copycols` variable as it is not needed now. Then I will set `source_col_used` to `true` when I initialize it if `copycols=true` is passed.